Dependency on internal AGP API breaks on AGP 3.4

Tested updating to AGP 3.4.0-alpha01 today in https://github.com/hzsweers/catchup

Happens on configuration

```
Caused by: java.lang.NoClassDefFoundError: com/android/build/gradle/internal/core/Toolchain
        at com.bugsnag.android.gradle.BugsnagPlugin.getCmakeToolchain(BugsnagPlugin.groovy:352)
        at com.bugsnag.android.gradle.BugsnagPlugin.setupNdkMappingFileUpload(BugsnagPlugin.groovy:149)
        at com.bugsnag.android.gradle.BugsnagPlugin$_applyBugsnagToVariant_closure8.doCall(BugsnagPlugin.groovy:125)
        at com.bugsnag.android.gradle.BugsnagPlugin.applyBugsnagToVariant(BugsnagPlugin.groovy:117)
        at org.gradle.internal.metaobject.BeanDynamicObject$GroovyObjectAdapter.invokeOpaqueMethod(BeanDynamicObject.java:579)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:506)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:78)
        at com.bugsnag.android.gradle.BugsnagPlugin$_apply_closure1$_closure15.doCall(BugsnagPlugin.groovy:52)
        at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:71)
        at org.gradle.util.ConfigureUtil.configureTarget(ConfigureUtil.java:155)
        at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:106)
        at org.gradle.util.ConfigureUtil$WrappedConfigureAction.execute(ConfigureUtil.java:167)
        at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:158)
        at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:186)
        at org.gradle.api.DomainObjectCollection$all$0.call(Unknown Source)
        at com.bugsnag.android.gradle.BugsnagPlugin$_apply_closure1.doCall(BugsnagPlugin.groovy:51)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingClosure$1$1.run(DefaultListenerBuildOperationDecorator.java:186)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.reapply(DefaultUserCodeApplicationContext.java:58)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingClosure$1.run(DefaultListenerBuildOperationDecorator.java:181)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingClosure.doCall(DefaultListenerBuildOperationDecorator.java:178)
        at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:40)
        at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:25)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
        at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:230)
        at org.gradle.internal.event.BroadcastDispatc
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+## 3.X.X (TBD)
+
+### Bug fixes
+
+* Manually calculate location of objdump in gradle plugin
+[#136](https://github.com/bugsnag/bugsnag-android-gradle-plugin/pull/136)
+
 ## 3.5.0 (2018-10-18)
 
 ### Bug fixes

--- a/features/fixtures/ndkapp/app/build.gradle
+++ b/features/fixtures/ndkapp/app/build.gradle
@@ -13,6 +13,9 @@ android {
                 cppFlags ""
             }
         }
+        ndk {
+            abiFilters "x86", "x86_64", "armeabi", "armeabi-v7a", "arm64-v8a"
+        }
     }
     buildTypes {
         release {
@@ -25,25 +28,10 @@ android {
             path "CMakeLists.txt"
         }
     }
-
-    flavorDimensions "regular"
-
-    productFlavors {
-        x86 {
-            ndk {
-                abiFilter "x86"
-            }
-        }
-        arm {
-            ndk {
-                abiFilters "armeabi-v7a"
-            }
-        }
-    }
 }
 
 dependencies {
-    compile 'com.bugsnag:bugsnag-android-ndk:1.+'
+    compile 'com.bugsnag:bugsnag-android-ndk:4.+'
 }
 
 apply plugin: 'com.bugsnag.android.gradle'

--- a/features/fixtures/ndkapp/app/build.gradle
+++ b/features/fixtures/ndkapp/app/build.gradle
@@ -42,8 +42,16 @@ bugsnag {
     releasesEndpoint = "http://localhost:${System.env.MOCK_API_PORT}"
 
     def customPath = System.env.PROJECT_ROOT
+    def objdumpLocation = System.env.OBJDUMP_LOCATION
 
     if (customPath != null) {
         projectRoot = customPath
+    }
+
+    if (objdumpLocation != null) {
+        objdumpPaths = [
+                "x86": objdumpLocation,
+                "arm64-v8a": objdumpLocation
+        ]
     }
 }

--- a/features/ndk_app.feature
+++ b/features/ndk_app.feature
@@ -14,27 +14,27 @@ Scenario: NDK apps send requests
     And the request 1 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 1
     And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 1
-    And the payload field "arch" equals "armeabi-v7a" for request 1
+    And the payload field "arch" equals "armeabi" for request 1
 
     And the request 2 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 2
     And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 2
-    And the payload field "arch" equals "x86" for request 2
+    And the payload field "arch" equals "armeabi-v7a" for request 2
 
     And the request 3 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 3
     And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 3
-    And the payload field "arch" equals "arm64-v8a" for request 3
+    And the payload field "arch" equals "x86_64" for request 3
 
     And the request 4 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 4
     And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 4
-    And the payload field "arch" equals "armeabi" for request 4
+    And the payload field "arch" equals "arm64-v8a" for request 4
 
     And the request 5 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 5
     And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 5
-    And the payload field "arch" equals "x86_64" for request 5
+    And the payload field "arch" equals "x86" for request 5
 
     And the request 6 is valid for the Android Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 6
@@ -58,3 +58,18 @@ Scenario: Custom projectRoot is added to payload
 
     And the request 5 is valid for the Android NDK Mapping API
     And the payload field "projectRoot" equals "/repos/custom/my-app" for request 5
+
+# Sets a non-existent objdump location for x86 and arm64-v8a, delivery should proceed as normal for other files
+Scenario: Custom objdump location
+    When I set environment variable "OBJDUMP_LOCATION" to "/fake/objdump"
+    When I build the NDK app
+    Then I should receive 5 requests
+
+    And the request 1 is valid for the Android NDK Mapping API
+    And the payload field "arch" equals "armeabi" for request 1
+
+    And the request 2 is valid for the Android NDK Mapping API
+    And the payload field "arch" equals "armeabi-v7a" for request 2
+
+    And the request 3 is valid for the Android NDK Mapping API
+    And the payload field "arch" equals "x86_64" for request 3

--- a/features/ndk_app.feature
+++ b/features/ndk_app.feature
@@ -2,7 +2,7 @@ Feature: Plugin integrated in NDK app
 
 Scenario: NDK apps send requests
     When I build the NDK app
-    Then I should receive 6 requests
+    Then I should receive 7 requests
 
     And the request 0 is valid for the Build API
     And the payload field "appVersion" equals "1.0" for request 0
@@ -11,34 +11,50 @@ Scenario: NDK apps send requests
     And the payload field "buildTool" equals "gradle-android" for request 0
     And the payload field "appVersionCode" equals "1" for request 0
 
-    And the request 1 is valid for the Build API
-    And the payload field "appVersion" equals "1.0" for request 1
+    And the request 1 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 1
-    And the payload field "builderName" is not null for request 1
-    And the payload field "buildTool" equals "gradle-android" for request 1
-    And the payload field "appVersionCode" equals "1" for request 1
+    And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 1
+    And the payload field "arch" equals "armeabi-v7a" for request 1
 
-    And the request 2 is valid for the Android Mapping API
+    And the request 2 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 2
+    And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 2
+    And the payload field "arch" equals "x86" for request 2
 
     And the request 3 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 3
     And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 3
+    And the payload field "arch" equals "arm64-v8a" for request 3
 
     And the request 4 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 4
     And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 4
+    And the payload field "arch" equals "armeabi" for request 4
 
-    And the request 5 is valid for the Android Mapping API
+    And the request 5 is valid for the Android NDK Mapping API
     And the payload field "apiKey" equals "your-api-key-here" for request 5
+    And the payload field "projectRoot" ends with "features/fixtures/ndkapp/app" for request 5
+    And the payload field "arch" equals "x86_64" for request 5
+
+    And the request 6 is valid for the Android Mapping API
+    And the payload field "apiKey" equals "your-api-key-here" for request 6
 
 Scenario: Custom projectRoot is added to payload
     When I set environment variable "PROJECT_ROOT" to "/repos/custom/my-app"
     When I build the NDK app
-    Then I should receive 6 requests
+    Then I should receive 7 requests
+
+    And the request 1 is valid for the Android NDK Mapping API
+    And the payload field "projectRoot" equals "/repos/custom/my-app" for request 1
+
+    And the request 2 is valid for the Android NDK Mapping API
+    And the payload field "projectRoot" equals "/repos/custom/my-app" for request 2
 
     And the request 3 is valid for the Android NDK Mapping API
     And the payload field "projectRoot" equals "/repos/custom/my-app" for request 3
 
     And the request 4 is valid for the Android NDK Mapping API
     And the payload field "projectRoot" equals "/repos/custom/my-app" for request 4
+
+    And the request 5 is valid for the Android NDK Mapping API
+    And the payload field "projectRoot" equals "/repos/custom/my-app" for request 5

--- a/src/main/groovy/com/bugsnag/android/gradle/Abi.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/Abi.groovy
@@ -1,0 +1,49 @@
+package com.bugsnag.android.gradle
+
+enum Abi {
+
+    ARMEABI(
+        "armeabi",
+        "arm-linux-androideabi",
+        "arm-linux-androideabi"
+    ),
+    ARMEABI_V7A(
+        "armeabi-v7a",
+        "arm-linux-androideabi",
+        "arm-linux-androideabi"
+    ),
+    ARM64_V8A(
+        "arm64-v8a",
+        "aarch64-linux-android",
+        "aarch64-linux-android"
+    ),
+    X86(
+        "x86",
+        "x86",
+        "i686-linux-android"
+    ),
+    X86_64(
+        "x86_64",
+        "x86_64",
+        "x86_64-linux-android"
+    );
+
+    final String abiName
+    final String toolchainPrefix
+    final String objdumpPrefix
+
+    Abi(String abiName, String toolchainPrefix, String objdumpPrefix) {
+        this.abiName = abiName
+        this.toolchainPrefix = toolchainPrefix
+        this.objdumpPrefix = objdumpPrefix
+    }
+
+    static Abi findByName(String abiName) {
+        for (Abi value : values()) {
+            if (value.abiName == abiName) {
+                return value
+            }
+        }
+        return null
+    }
+}

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy
@@ -6,7 +6,6 @@ import com.android.build.gradle.api.ApplicationVariant
 import com.android.build.gradle.api.BaseVariant
 import com.android.build.gradle.api.BaseVariantOutput
 import com.android.build.gradle.api.LibraryVariant
-import com.android.build.gradle.internal.core.Toolchain
 import com.android.build.gradle.internal.dsl.BuildType
 import org.gradle.api.DomainObjectSet
 import org.gradle.api.Plugin
@@ -146,7 +145,6 @@ class BugsnagPlugin implements Plugin<Project> {
             uploadNdkTask.variantName = taskNameForVariant(deps.variant)
             uploadNdkTask.projectDir = project.projectDir
             uploadNdkTask.rootDir = project.rootDir
-            uploadNdkTask.toolchain = getCmakeToolchain(project, deps.variant)
             uploadNdkTask.sharedObjectPath = project.bugsnag.sharedObjectPath
         }
     }
@@ -310,63 +308,6 @@ class BugsnagPlugin implements Plugin<Project> {
         } else {
             return false
         }
-    }
-
-    /**
-     * Gets the buildchain that is setup for cmake
-     * @param project The project to check
-     * @param variant The variant to check
-     * @return The buildchain for cmake (or Toolchain.default if not found)
-     */
-    private static String getCmakeToolchain(Project project, BaseVariant variant) {
-        String toolchain = null
-
-        // First check the selected build type to see if there are cmake arguments
-        def buildTypes = project.android.buildTypes.store
-        BuildType b = findNode(buildTypes, variant.baseName)
-
-        if (b != null
-            && b.externalNativeBuildOptions != null
-            && b.externalNativeBuildOptions.cmake != null
-            && b.externalNativeBuildOptions.cmake.arguments != null) {
-
-            ArrayList<String> args = b.externalNativeBuildOptions.cmake.arguments
-            toolchain = getToolchain(args)
-        }
-
-        // Next check to see if there are arguments in the default config section
-        if (toolchain == null) {
-            if (project.android.defaultConfig.externalNativeBuildOptions != null
-                && project.android.defaultConfig.externalNativeBuildOptions.cmake != null
-                && project.android.defaultConfig.externalNativeBuildOptions.cmake.arguments != null) {
-
-                ArrayList<String> args = project.android.defaultConfig.externalNativeBuildOptions.cmake.arguments
-                for (String arg : args) {
-                    toolchain = getToolchain(args)
-                }
-            }
-        }
-
-        // Default to Toolchain.default if not found so far
-        if (toolchain == null) {
-            toolchain = Toolchain.default.name
-        }
-
-        return toolchain
-    }
-
-    /**
-     * Looks for an "ANDROID_TOOLCHAIN" argument in a list of cmake arguments
-     * @param args The cmake args
-     * @return the value of the "ANDROID_TOOLCHAIN" argument, or null if not found
-     */
-    private static String getToolchain(ArrayList<String> args) {
-        for (String arg : args) {
-            if (arg.startsWith("-DANDROID_TOOLCHAIN")) {
-                return arg.substring(arg.indexOf("=") + 1).trim()
-            }
-        }
-        return null
     }
 
     /**

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagPluginExtension.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagPluginExtension.groovy
@@ -21,6 +21,7 @@ class BugsnagPluginExtension {
     // release API values
     String builderName = null
     Map<String, String> metadata = null
+    Map<String, String> objdumpPaths = null
 
 }
 

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
@@ -6,6 +6,8 @@ import com.android.build.gradle.tasks.ProcessAndroidResources
 import org.apache.http.entity.mime.MultipartEntity
 import org.apache.http.entity.mime.content.FileBody
 import org.apache.http.entity.mime.content.StringBody
+import org.apache.tools.ant.taskdefs.condition.Os
+import org.gradle.api.Project
 import org.gradle.api.tasks.TaskAction
 
 import java.util.zip.GZIPOutputStream
@@ -204,13 +206,44 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
      * @return The objdump executable, or null if not found
      */
     File getObjDumpExecutable(String arch) {
-
         try {
-            return objDumpPath
+            File objDumpFile = findObjDump(project, arch)
+
+            if (!objDumpFile.exists() || !objDumpFile.canExecute()) {
+                throw new RuntimeException("Failed to find executable objdump at $objDumpFile")
+            }
+            return objDumpFile
         } catch (Throwable ex) {
             project.logger.error("Error attempting to calculate objdump location: " + ex.message)
         }
-
         return null
     }
+
+    static File findObjDump(Project project, String arch) {
+        Abi abi = Abi.findByName(arch)
+        String ndkDir = project.android.ndkDirectory
+        String osName = calculateOsName()
+        return calculateObjDumpLocation(ndkDir, abi, osName)
+    }
+
+    static File calculateObjDumpLocation(String ndkDir, Abi abi, String osName) {
+        new File("$ndkDir/toolchains/$abi.toolchainPrefix-4.9/prebuilt/$osName/bin/$abi.objdumpPrefix-objdump")
+    }
+
+    static String calculateOsName() {
+        if (Os.isFamily(Os.FAMILY_MAC)) {
+            return "darwin-x86_64"
+        } else if (Os.isFamily(Os.FAMILY_UNIX)) {
+            return "linux-x86_64"
+        } else if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+            if ("x86" == System.getProperty("os.arch")) { // 32-bit
+                return "windows"
+            } else {
+                return "windows-x86_64"
+            }
+        } else {
+            return "unknown"
+        }
+    }
+
 }

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
@@ -207,7 +207,14 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
      */
     File getObjDumpExecutable(String arch) {
         try {
-            File objDumpFile = findObjDump(project, arch)
+            String override = getObjDumpOverride(arch)
+            File objDumpFile
+
+            if (override != null) {
+                objDumpFile = new File(override)
+            } else {
+                objDumpFile = findObjDump(project, arch)
+            }
 
             if (!objDumpFile.exists() || !objDumpFile.canExecute()) {
                 throw new RuntimeException("Failed to find executable objdump at $objDumpFile")
@@ -217,6 +224,11 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
             project.logger.error("Error attempting to calculate objdump location: " + ex.message)
         }
         return null
+    }
+
+    private Object getObjDumpOverride(String arch) {
+        Map<String, String> paths = project.bugsnag.objdumpPaths
+        return paths != null ? paths[arch] : null
     }
 
     static File findObjDump(Project project, String arch) {

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
@@ -235,6 +235,13 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
         Abi abi = Abi.findByName(arch)
         String ndkDir = project.android.ndkDirectory
         String osName = calculateOsName()
+
+        if (abi == null) {
+            throw new IllegalStateException("Failed to find ABI for $arch")
+        }
+        if (osName == null) {
+            throw new IllegalStateException("Failed to calculate OS name")
+        }
         return calculateObjDumpLocation(ndkDir, abi, osName)
     }
 
@@ -254,7 +261,7 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
                 return "windows-x86_64"
             }
         } else {
-            return "unknown"
+            return null
         }
     }
 

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy
@@ -1,8 +1,6 @@
 package com.bugsnag.android.gradle
 
 import com.android.build.gradle.api.BaseVariantOutput
-import com.android.build.gradle.internal.core.Abi
-import com.android.build.gradle.internal.ndk.NdkHandler
 import com.android.build.gradle.tasks.ExternalNativeBuildTask
 import com.android.build.gradle.tasks.ProcessAndroidResources
 import org.apache.http.entity.mime.MultipartEntity
@@ -33,7 +31,6 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
     String variantName
     File projectDir
     File rootDir
-    String toolchain
     String sharedObjectPath
 
     BugsnagUploadNdkTask() {
@@ -209,9 +206,6 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
     File getObjDumpExecutable(String arch) {
 
         try {
-            Abi abi = Abi.getByName(arch)
-            NdkHandler handler = new NdkHandler(rootDir, null, toolchain, "", true)
-            File objDumpPath = new File(handler.getDefaultGccToolchainPath(abi), "bin/" + abi.getGccExecutablePrefix() + "-objdump")
             return objDumpPath
         } catch (Throwable ex) {
             project.logger.error("Error attempting to calculate objdump location: " + ex.message)

--- a/src/test/groovy/com/bugsnag/android/gradle/ObjDumpLocationTest.groovy
+++ b/src/test/groovy/com/bugsnag/android/gradle/ObjDumpLocationTest.groovy
@@ -32,7 +32,7 @@ class ObjDumpLocationTest {
         return inputs
     }
 
-    @Parameterized.Parameters(name = "1")
+    @Parameterized.Parameters
     static Collection<String> os() {
         return Arrays.asList("windows", "linux")
     }

--- a/src/test/groovy/com/bugsnag/android/gradle/ObjDumpLocationTest.groovy
+++ b/src/test/groovy/com/bugsnag/android/gradle/ObjDumpLocationTest.groovy
@@ -1,0 +1,47 @@
+package com.bugsnag.android.gradle
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+import static org.junit.Assert.*
+
+@RunWith(Parameterized.class)
+class ObjDumpLocationTest {
+
+    @Parameterized.Parameter(0)
+    public Abi abi
+
+    @Parameterized.Parameter(1)
+    public String osName
+
+    @Parameterized.Parameter(2)
+    public String ndkDir
+
+    @Parameterized.Parameters
+    static Collection<Object[]> inputs() {
+        Collection<Object> inputs = new ArrayList<>();
+
+        for (Abi abi: Abi.values()) {
+            for (String os: ["darwin-x86_64", "linux-x86_64", "windows", "windows-x86_64"]) {
+                for (String ndkDir: ["/Users/bob/Library/Android/sdk/ndk-bundle", "/etc/ndk-bundle"]) {
+                    inputs.add([abi, os, ndkDir].toArray())
+                }
+            }
+        }
+        return inputs
+    }
+
+    @Parameterized.Parameters(name = "1")
+    static Collection<String> os() {
+        return Arrays.asList("windows", "linux")
+    }
+
+    @Test
+    void defaultObjDumpLocation() {
+        File file = BugsnagUploadNdkTask.calculateObjDumpLocation(ndkDir, abi, osName)
+        String expected = "$ndkDir/toolchains/$abi.toolchainPrefix-4.9/prebuilt/$osName/bin/$abi.objdumpPrefix-objdump"
+        assertEquals(expected, file.path)
+    }
+
+}

